invenio_communities/communities/services/components.py

```diff
@@ -300,6 +300,23 @@ def update(self, identity, data=None, record=None, errors=None, **kwargs):
                 record["theme"] = data["theme"]
 
 
+class ChildrenComponent(ServiceComponent):
+    """Service component for children integration."""
+
+    def update(self, identity, data=None, record=None, **kwargs):
+        """Update handler."""
+        # We check if the children field is passed and it's different to the values stored
+        if "children" in data and data["children"] != record.get("children", {}):
+            self.service.require_permission(identity, "manage_children", record=record)
+            record["children"].update(data["children"])
+
+    def create(self, identity, data=None, record=None, **kwargs):
+        """Create handler."""
+        if "children" in data:
+            self.service.require_permission(identity, "manage_children", record=record)
+            record["children"] = data["children"]
```

Comment:
nit: to preserve the logical/original order this method should come before `update`